lahja/asyncio/endpoint.py

```diff
@@ -286,17 +288,25 @@ class AsyncioEndpoint(BaseEndpoint):
     _sync_handler: DefaultDict[Type[BaseEvent], List[SubscriptionSyncHandler]]
 
     _loop: Optional[asyncio.AbstractEventLoop] = None
+    _get_request_id: Iterator[RequestID]
 
     def __init__(self, name: str) -> None:
         self.name = name
 
+        try:
+            self._get_request_id = RequestIDGenerator(name.encode("ascii") + b":")
```

Comment:
nitpick: In an imaginary future where there are multiple backends users might want to switch between different kinds of endpoints. If this check is on the Asyncio endpoint but not the Trio one then non-ascii users could have a hard time experimenting with Asyncio after building their app with Trio.

Putting this check into `BaseEndpoint` would make life easier for those hypothetical users.

Reply:
:+1: